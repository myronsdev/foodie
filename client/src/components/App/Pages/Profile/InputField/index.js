import React from 'react'
import PropTypes from 'prop-types'

const InputField = (props) => {
  const { value, onChange, name } = props

  let title = ''
  let type = ''

  switch (name) {
    case 'recipeTitle':
      title = 'Title'
      break
    case 'description':
      title = 'Description'
      break
    case 'cookTime':
      title = 'Cook Time'
      type = 'number'
      break
    case 'prepTime':
      title = 'Prep Time'
      type = 'number'
      break
    case 'serves':
      title = 'Serves'
      type = 'number'
      break
    case 'directions':
      title = 'Directions'
      break
    default:
      title = ''
      type = 'text'
  }

  return (
    <div className="field">
      <label className="label">{title}</label>
      <div className="control">
        <input className="input" name={name} type={type} value={value} onChange={onChange} />
      </div>
    </div>
  )
}

InputField.propTypes = {
  value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  onChange: PropTypes.func.isRequired,
  name: PropTypes.string.isRequired
}

export default InputField
