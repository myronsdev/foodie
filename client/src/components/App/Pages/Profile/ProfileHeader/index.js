import React from 'react'
import PropTypes from 'prop-types'

const ProfileHeader = (props) => {
  console.log(props)
  return <p className="user-greeting">This is your profile page, {props.name}</p>
}

ProfileHeader.propTypes = {
  name: PropTypes.string.isRequired
}

export default ProfileHeader
