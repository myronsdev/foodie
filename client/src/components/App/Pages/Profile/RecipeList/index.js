import React from 'react'
import PropTypes from 'prop-types'
import { Link } from 'react-router-dom'

const RecipeList = (props) => {
  console.log(props)
  const recipes = props.recipes

  let listRecipes = []
  if (recipes) {
    listRecipes = recipes.map((recipe) => (
      <li key={recipe._id}>
        <Link to={`/profile/recipe/` + recipe._id}>{recipe.title}</Link>
      </li>
    ))
  }

  return <ul>{listRecipes}</ul>
}

RecipeList.propTypes = {
  recipes: PropTypes.array.isRequired
}

export default RecipeList
