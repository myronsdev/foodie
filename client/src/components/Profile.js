import React, { Component } from 'react'

class Profile extends Component {
  render() {
    if (this.props.loading === true) {
      return <p />
    }
    if (this.props.loading === false) {
      return <p>This is your profile page, {this.props.user.name}</p>
    }
  }
}

export default Profile
