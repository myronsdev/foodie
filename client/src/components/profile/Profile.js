import React, { Component } from 'react'
import axios from 'axios'

class Profile extends Component {
  async componentDidUpdate() {
    try {
      const res = await axios.get('/api/profile/recipes')
      console.log(res)
    } catch (e) {
      console.log(e)
    }
  }

  renderContent() {
    if (this.props.loading === true) {
      return <p />
    }
    if (this.props.loading === false) {
      return <p>This is your profile page, {this.props.user.name}</p>
    }
  }

  render() {
    return (
      <section className="section">
        <div className="container is-fluid">
          <div className="columns">
            <div className="column">{this.renderContent()}</div>
            <div className="column">
              <h2>MY RECIPES</h2>
            </div>
          </div>
        </div>
      </section>
    )
  }
}

export default Profile
