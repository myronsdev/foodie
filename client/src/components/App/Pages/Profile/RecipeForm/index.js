import React, { Component } from 'react'
import Checkbox from '../AddRecipe/Checkbox'
import { Redirect } from 'react-router-dom'
import InputField from '../InputField'
import axios from 'axios'

class RecipeForm extends Component {
  state = {
    recipeTitle: '',
    description: '',
    foodType: [],
    cookTime: 0,
    prepTime: 0,
    ingredients: [],
    serves: 0,
    directions: '',
    allowedFoodTypes: [],
    sweeteners: [],
    vegetables: [],
    meats: [],
    seasoning: [],
    fruits: [],
    dairy: [],
    grains: [],
    oils: [],
    redirect: false
  }
  async componentDidMount() {
    const res = await axios.get('/api/formData')

    this.setState({ allowedFoodTypes: res.data.foodTypes })
    delete res.data.foodTypes

    Object.keys(res.data).forEach((key) => {
      this.setState(() => ({ [key]: res.data[key] }))
    })
  }

  handleSubmit = async (event) => {
    event.preventDefault()
    try {
      const res = await axios({
        method: 'post',
        url: '/api/recipe',
        data: {
          title: this.state.recipeTitle,
          description: this.state.description,
          foodType: this.state.foodType,
          cookTime: this.state.cookTime,
          prepTime: this.state.prepTime,
          ingredients: this.state.ingredients,
          rating: 2,
          serves: this.state.serves,
          directions: this.state.directions
        }
      })
      if (res) {
        this.setState(() => ({ redirect: true }))
      }
    } catch (e) {
      console.log(e)
    }
  }

  handleChange = (event) => {
    const target = event.target
    const value = target.value
    const name = target.name

    this.setState({ [name]: value })
  }

  handlefoodTypeCheckboxChange = (event) => {
    const target = event.target.name

    if (!this.state.foodType.includes(target)) {
      this.setState((prevState) => ({
        foodType: [...prevState.foodType, target]
      }))
    } else {
      this.setState((prevState) => ({
        foodType: [...prevState.foodType.filter((current) => current !== target)]
      }))
    }
  }

  handleIngredientsCheckboxChange = (event) => {
    const target = event.target.name

    if (!this.state.ingredients.includes(target)) {
      this.setState((prevState) => ({
        ingredients: [...prevState.ingredients, target]
      }))
    } else {
      this.setState((prevState) => ({
        ingredients: [...prevState.ingredients.filter((current) => current !== target)]
      }))
    }
  }

  showFoodTypeChecked(searchFoodType) {
    if (this.state.foodType.includes(searchFoodType)) {
      return true
    }
  }

  showIngredientChecked(searchIngredient) {
    if (this.state.ingredients.includes(searchIngredient)) {
      return true
    }
  }
  render() {
    if (this.state.redirect) {
      return <Redirect to="/profile" />
    }
    return (
      <form onSubmit={this.handleSubmit}>
        <InputField
          name={Object.getOwnPropertyNames(this.state)[0]}
          value={this.state.recipeTitle}
          onChange={this.handleChange}
        />
        <InputField
          name={Object.getOwnPropertyNames(this.state)[1]}
          value={this.state.description}
          onChange={this.handleChange}
        />
        <div className="field content">
          <h4>Food Types</h4>
          <div className="control is-grouped">
            {this.state.allowedFoodTypes.map((item) => (
              <Checkbox
                key={item}
                type="checkbox"
                name={item}
                checked={this.showFoodTypeChecked(item)}
                onChange={this.handlefoodTypeCheckboxChange}
              />
            ))}
          </div>
        </div>
        <InputField
          name={Object.getOwnPropertyNames(this.state)[3]}
          value={this.state.cookTime}
          onChange={this.handleChange}
        />
        <InputField
          name={Object.getOwnPropertyNames(this.state)[4]}
          value={this.state.prepTime}
          onChange={this.handleChange}
        />
        <div className="field content">
          <h4>Vegetables</h4>
          <div className="control is-grouped">
            {this.state.vegetables.map((item) => (
              <Checkbox
                key={item}
                type="checkbox"
                name={item}
                checked={this.showIngredientChecked(item)}
                onChange={this.handleIngredientsCheckboxChange}
              />
            ))}
          </div>
        </div>
        <div className="field content">
          <h4>Fruits</h4>
          <div className="control is-grouped">
            {this.state.fruits.map((item) => (
              <Checkbox
                key={item}
                type="checkbox"
                name={item}
                checked={this.showIngredientChecked(item)}
                onChange={this.handleIngredientsCheckboxChange}
              />
            ))}
          </div>
        </div>
        <div className="field content">
          <h4>Seasoning</h4>
          <div className="control is-grouped">
            {this.state.seasoning.map((item) => (
              <Checkbox
                key={item}
                type="checkbox"
                name={item}
                checked={this.showIngredientChecked(item)}
                onChange={this.handleIngredientsCheckboxChange}
              />
            ))}
          </div>
        </div>
        <div className="field content">
          <h4>Sweeteners</h4>
          <div className="control is-grouped">
            {this.state.sweeteners.map((item) => (
              <Checkbox
                key={item}
                type="checkbox"
                name={item}
                checked={this.showIngredientChecked(item)}
                onChange={this.handleIngredientsCheckboxChange}
              />
            ))}
          </div>
        </div>
        <div className="field content">
          <h4>Meats</h4>
          <div className="control is-grouped">
            {this.state.meats.map((item) => (
              <Checkbox
                key={item}
                type="checkbox"
                name={item}
                checked={this.showIngredientChecked(item)}
                onChange={this.handleIngredientsCheckboxChange}
              />
            ))}
          </div>
        </div>
        <div className="field content">
          <h4>Dairy</h4>
          <div className="control is-grouped">
            {this.state.dairy.map((item) => (
              <Checkbox
                key={item}
                type="checkbox"
                name={item}
                checked={this.showIngredientChecked(item)}
                onChange={this.handleIngredientsCheckboxChange}
              />
            ))}
          </div>
        </div>
        <div className="field content">
          <h4>Grains</h4>
          <div className="control is-grouped">
            {this.state.grains.map((item) => (
              <Checkbox
                key={item}
                type="checkbox"
                name={item}
                checked={this.showIngredientChecked(item)}
                onChange={this.handleIngredientsCheckboxChange}
              />
            ))}
          </div>
        </div>
        <div className="field content">
          <h4>Oils</h4>
          <div className="control is-grouped">
            {this.state.oils.map((item) => (
              <Checkbox
                key={item}
                type="checkbox"
                name={item}
                checked={this.showIngredientChecked(item)}
                onChange={this.handleIngredientsCheckboxChange}
              />
            ))}
          </div>
        </div>
        <InputField
          name={Object.getOwnPropertyNames(this.state)[6]}
          value={this.state.serves}
          onChange={this.handleChange}
        />
        <InputField
          name={Object.getOwnPropertyNames(this.state)[7]}
          value={this.state.directions}
          onChange={this.handleChange}
        />
        <input className="submit button is-link" type="submit" value="Submit" />
      </form>
    )
  }
}

export default RecipeForm
