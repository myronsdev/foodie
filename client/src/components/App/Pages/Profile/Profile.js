import React, { Component, Fragment } from 'react'
import ProfileHeader from './ProfileHeader'
import RecipeList from './RecipeList'
import { Link } from 'react-router-dom'
import axios from 'axios'

class Profile extends Component {
  constructor(props) {
    super(props)
    this.state = {
      recipes: []
    }
  }

  async componentDidMount() {
    try {
      const res = await axios.get('/api/profile/recipes')

      this.setState(() => ({ recipes: res.data }))
    } catch (e) {
      console.log(e)
    }
  }

  render() {
    const { name } = this.props.user
    const { recipes } = this.state
    const loading = recipes.length === 0

    return (
      <Fragment>
        <div className="column">
          <div className="content">{name ? <ProfileHeader name={name} /> : <div />}</div>
        </div>
        <div className="column">
          <div className="content">
            <h2>MY RECIPES</h2>
            {!loading ? <RecipeList recipes={recipes} /> : <div />}
            <Link className="button is-link" to="/profile/recipe">
              Add New
            </Link>
          </div>
        </div>
      </Fragment>
    )
  }
}

export default Profile
