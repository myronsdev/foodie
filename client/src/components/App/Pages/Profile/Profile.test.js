import React from 'react'
import Profile from './Profile'
import { BrowserRouter as Router } from 'react-router-dom'
import renderer from 'react-test-renderer'

it('renders correctly', () => {
  const tree = renderer.create(
    <Router>
      <Profile />
    </Router>
  )

  expect(tree).toMatchSnapshot()
})
