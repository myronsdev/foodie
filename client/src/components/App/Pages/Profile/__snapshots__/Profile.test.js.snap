// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <div
    className="column"
  />,
  <div
    className="column"
  >
    <h2>
      MY RECIPES
    </h2>
    <ul />
    <a
      href="/profile/recipe"
      onClick={[Function]}
    >
      Add New
    </a>
  </div>,
]
`;
